composer/src/Plugin.php

```diff
@@ -0,0 +1,63 @@
+<?php
+/**
+ * @file
+ * Contains Beet\Box|Plugin.
+ */
+
+namespace Beet\Box;
+
+use Composer\Composer;
+use Composer\EventDispatcher\EventSubscriberInterface;
+use Composer\Factory;
+use Composer\IO\IOInterface;
+use Composer\Plugin\PluginInterface;
+use Composer\Script\Event;
+use Composer\Script\ScriptEvents;
+
+class Plugin implements PluginInterface, EventSubscriberInterface {
+
+    /**
+     * @var \Composer\Composer
+     */
+    protected $composer;
+
+    /**
+     * @var \Composer\IO\IOInterface
+     */
+    protected $io;
+
+    /**
+     * {@inheritdoc}
+     */
+    public function activate(Composer $composer, IOInterface $io) {
+        $this->composer = $composer;
+        $this->io = $io;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public static function getSubscribedEvents() {
+        return array(
+            ScriptEvents::POST_INSTALL_CMD => 'addVagrantfile',
+            ScriptEvents::POST_UPDATE_CMD => 'addVagrantfile',
+        );
+    }
+
+    /**
+     * Add/update project Vagrantfile.
+     *
+     * @param \Composer\Script\Event $event
+     */
+    public function addVagrantfile(Event $event) {
+
+        $baseDir = dirname(Factory::getComposerFile());
+        $source = __DIR__ . '/../../.beetbox/Vagrantfile';
+        $target =  $baseDir . '/Vagrantfile';
+
+        if (!file_exists($target) || md5_file($source) != md5_file($target)) {
```

Comment:
I originally had seperate methods for create and update with a message for each but decided to go with a combined method. 

My thinking was that it shouldn't matter if the the Vagrantfile was updated or not as you'd already see that the package was updated via the `composer.lock`